force-app/main/default/classes/SOQL.cls

```diff
@@ -63,9 +80,24 @@ public with sharing class SOQL {
      * @return A list of Opportunities with important opportunities fields
      * Update the corresponding test class and testGetClosedWonOpportunityCountByAccount method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static List<Opportunity> getClosedWonOpportunities(Id accountId) {
+        List<Opportunity> closedWonOpptys = new List<Opportunity>();
+
+        List<Account> accountsWithClosedWonOpptys = [SELECT Id, (SELECT Id, Name, StageName, CloseDate, AccountId, Amount
+                                                                    FROM Opportunities
+                                                                    WHERE StageName = 'Closed Won') 
+                                                        FROM Account
+                                                        WHERE Id = :accountId];
+
+        for(Account acct : accountsWithClosedWonOpptys) {
+            for (Opportunity oppty : acct.Opportunities) {
+                closedWonOpptys.add(oppty);
+            }
+        }
+        
```

Comment:
Take a look at the formatting and approach here. You only need ONE query to get this done, and no nested for loops.

```suggestion
        List<Opportunity> opps = [
                SELECT 
                        Id,
                        Name,
                        StageName,
                        AccountId,
                        CloseDate,
                        Amount 
                FROM Opportunity 
                WHERE AccountId = :accId 
                AND StageName = 'Closed Won'
        ];
```